src/editor/tools/info_tool.cc

```diff
@@ -19,223 +19,43 @@
-		caps_strings.push_back(_("walkable"));
-	}
-	if (caps & Widelands::MOVECAPS_SWIM) {
-		/** TRANSLATORS: Editor terrain property: units can swim on this terrain (fish, ships) */
-		caps_strings.push_back(_("swimmable"));
-	}
-
-	buf += as_listitem(
-	   bformat(_("Caps: %s"), i18n::localize_list(caps_strings, i18n::ConcatenateWith::COMMA)),
-	   font_style);
-
-	if (f.get_owned_by() > 0) {
-		buf +=
-		   as_listitem(bformat(_("Owned by: Player %u"), static_cast<unsigned int>(f.get_owned_by())),
-		               font_style);
-	} else {
-		buf += as_listitem(_("Owned by: —"), font_style);
-	}
-
-	// *** Terrain info
-	buf += as_heading(_("Terrain"), UI::PanelStyle::kWui);
-
-	const Widelands::Field& tf = (*map)[center.triangle.node];
-	const Widelands::TerrainDescription* ter = parent.egbase().descriptions().get_terrain_descr(
-	   center.triangle.t == Widelands::TriangleIndex::D ? tf.terrain_d() : tf.terrain_r());
-
-	buf += as_listitem(bformat(pgettext("terrain_name", "Name: %s"), ter->descname()), font_style);
-
-	std::vector<std::string> terrain_is_strings;
-	for (const Widelands::TerrainDescription::Type& terrain_type : ter->get_types()) {
-		terrain_is_strings.push_back(terrain_type.descname);
-	}
-
-	buf += as_listitem(
-	   bformat(
-	      /** TRANSLATORS: "Is" is a list of terrain properties, e.g. "arable, unreachable and
-	       * unwalkable". You can also translate this as "Category: %s" or "Property: %s" */
-	      _("Is: %s"), i18n::localize_list(terrain_is_strings, i18n::ConcatenateWith::AMPERSAND)),
-	   font_style);
-
-	// *** Map Object info
-	const Widelands::BaseImmovable* immovable = f.get_immovable();
-	Widelands::Bob* bob = f.get_first_bob();
-	if (immovable || bob) {
-		/** TRANSLATORS: Heading for immovables and animals in editor info tool */
-		buf += as_heading(_("Objects"), UI::PanelStyle::kWui);
-		if (immovable) {
-			buf += as_listitem(bformat(_("Immovable: %s"), immovable->descr().descname()), font_style);
+	Widelands::Field& tf = (*map)[center.triangle.node];
+
+	UI::UniqueWindow::Registry& registry = parent.unique_windows().get_registry(
+	   bformat("fieldinfo_%d_%d", center.node.x, center.node.y));
+	registry.open_window = [this, &parent, &registry, &center, &f, &tf, map]() {
+		// if window reaches bottom right corner, start from top left corner again
+		int a = (parent.get_inner_w() - FieldInfoWindow::total_width) / kOffset;
+		int b = (parent.get_inner_h() - FieldInfoWindow::total_height) / kOffset;
+		int offset_factor = number_of_open_windows_ % std::min(a, b);
+		int offset = offset_factor * kOffset;
+		log_dbg("a: %d, b: %d, number: %d offsetfactor: %d, offset: %d", a, b,
+		        number_of_open_windows_.load(), offset_factor, offset);
+
+		new FieldInfoWindow(parent, registry, offset, offset, center, f, tf, map);
+		log_dbg("increment now: %d", number_of_open_windows_.load());
+		++number_of_open_windows_;
+	};
+	registry.closed.connect([this]() {
```

Comment:
I assume the error might be here. Every time this function is called because the player clicks on a field, you connect such a lambda to the registry's `closed` signal. If the player clicks on the same field *again*, you again connect this lambda even though the registry for this field already has a subscription. So there are now *two* lambdas attached to this registry's `closed` signal which *both* decrement the counter. The third time the player clicks the field the closing signal now has three lambdas, and so on. So you need to remember which registries already have a lambda, or replace the existing lambda when registering a new one. I'd recommend the second option: Use `subscribe` instead of `connect` and store its return value in a member variable of type `std::unique_ptr<...>`:
```c++
cached_subscriber_.reset(registry.closed.subscribe([this]() {
```